gen/src/MPIgenerator.jl

```diff
@@ -10,6 +11,39 @@ module MPIgenerator
         error("Unknown MPI binary: $(MPIPreferences.binary)")
     end
 
+    rewrite!(dag::ExprDAG) = begin
+        replace!(get_nodes(dag)) do node
+            signature = Clang.name(node.cursor)
+
+            if startswith(signature, "MPI_Type_create_keyval") || true
```

Comment:
This isn't the only one wrong, all pointers which aren't buffers should be changed

Reply:
Yes, I know, this is a draft.
I'm only opening this PR now to discuss Clang internals with @Gnimuc.